pkg/reference/reference_test.go

```diff
@@ -455,6 +535,30 @@ func TestResolveMultiple(t *testing.T) {
 				err: nil,
 			},
 		},
+		"DiscardTerminatingFromSelect": {
+			reason: "A managed resource should not resolve a deleted object",
+			c: &test.MockClient{
+				MockList: test.NewMockListFn(nil),
+			},
+			from: controlled,
+			args: args{
+				req: MultiResolutionRequest{
+					Selector: &xpv1.Selector{},
+					To: To{List: &FakeManagedList{Items: []resource.Managed{
+						&fake.Managed{ObjectMeta: metav1.ObjectMeta{Name: "Deleted", DeletionTimestamp: &now}}, // A resource that does not match.
```

Comment:
@hasheddan Isn't this the testcase?

Input:
- A (deletion requested by user)
- B (live)

Output:
- B

if not, can you write a bit longer explanation about the test case you're thinking about, and the expected outcome?